config/m4/cuda.m4

```diff
@@ -78,6 +81,8 @@ AS_IF([test "x$cuda_checked" != "xyes"],
                         [AC_DEFINE([HAVE_CUDA_FABRIC], 1, [Enable CUDA fabric handle support])],
                         [], [[#include <cuda.h>]])
 
+         AC_PATH_PROGS(NVCC, nvcc, "", $PATH:$CUDA_BIN_PATH)
```

Comment:
maybe
$CUDA_BIN_PATH:$PATH
to prefer the with arg over env var?

Reply:
Fixed.